Global Navigation + Sub Menu

darkdrgn2k opened the pull request:
Concept for issue #153 and #152 Creating Hover Drop Down menus 

Idea to create a Hierarchical structure that would be easy to navigate.
For Mobile/Tablet we would still need to include the sub items within the page itself some how
(perhaps a small right side menu)

Shrinks99 CP from #152 
- About
  - Timeline
  - Code of Conduct
- Events
- Get Involved
  - Build a Node*
  - Join the Network*
- Map*
- Contact
  - Twitter
  - Email
  - Riot
  - Github
* Indicates future pages

darkdrgn2k committed:
CSS Code for Sub Menus
LIQUID Code for Sub Menus
"Hard Coded" Contact drop down
Code Of Conduct moved into About sub menu
Get Involved moved further back
Time Line moved under About

garrying commented:
This is looking good so far. Were there thoughts on presenting the sub-menu in the mobile menu?

darkdrgn2k commented:
I'm thinking keep the mobile as is, maybe even hide the sub menu  but on the PARENT pages throw in another menu, (regardless of desktop or mobile), maybe a right side menu or something.

dcwalk commented:
Looking good, my only thought -- do we need the Contact with sub menu item?

darkdrgn2k commented:
Personally I like the idea cause it's a quick way to join the chat or jump into git with only one click.

But that's just me :)

Shrinks99 commented:
I like having a contact sub menu for the same reason already specified
above, unless we really need everyone to have more information about
contacting us and we want to enforce the fact that they will have to look
at that I would be in favour of keeping the sub menu with links to the
methods discussed on the main page.

TL;DR: yeah, that would be good to keep.

On 3 September 2017 at 20:19, darkdrgn2k <notifications@github.com> wrote:

> Personally I like the idea cause it's a quick way to join the chat or jump
> into git with only one click.
>
> But that's just me :)
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/tomeshnet/tomesh.net/pull/172#issuecomment-326841823>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFaPavEP6FLpc-lqvRa1zu06xOc8cn6Pks5se0IpgaJpZM4PHwDT>
> .
>



-- 
-Hank

Commit:
Added sub navigation for child objects

Commit:
Added hiding of menu on mobile pages without submenu

Commit:
Lowered Z Index for submenu not to interfere with top menu

Commit:
Increase padding Slightly for mobile pages

darkdrgn2k commented:
Sorry for all the commits.. done now let me know what you guys things of the submenu this way.

garrying committed:
Merge branch 'master' into GlobalNavigation

garrying commented:
@darkdrgn2k There are a few issues with the mobile layout:
- Menu items, regardless if there is a secondary menu, requires 2 clicks in order to navigate to the page
- Extra links are showing on the homepage:
![simulator screen shot sep 5 2017 10 25 39 pm](https://user-images.githubusercontent.com/227587/30091973-3e299ee4-928a-11e7-90f4-47c2c60e840d.png)
- The layout of the 'secondary' menu seems out of place—could be a styling issue:
![simulator screen shot sep 5 2017 10 26 19 pm](https://user-images.githubusercontent.com/227587/30091977-47940ca8-928a-11e7-8ea0-11e5d5fe1e27.png)
- secondary navigation doesn't persist when you go navigate to secondary pages

Commit:
Merge branch 'master' into GlobalNavigation

Commit:
Reordered Time Line and parented build a node to Get Involved

Commit:
Prevent displaying TOP LEVEL navigation as sub menu

Commit:
Added Same Level navigation into sub menu logic

darkdrgn2k commented:
> Extra links are showing on the homepage

Corrected in https://github.com/tomeshnet/tomesh.net/pull/172/commits/407c67edcb11380e5d232e95fdfe02e733d8ae74

> secondary navigation doesn't persist when you go navigate to secondary pages

Added Same Level Navigation
https://github.com/tomeshnet/tomesh.net/pull/172/commits/a0e4993e00e18e7c07f57622da07915f1c7ba122

darkdrgn2k commented:
> Menu items, regardless if there is a secondary menu, requires 2 clicks in order to navigate to the page

Does it really matter, the pages ARE related so i don't think its necessary for everything to be 1 click.
And even if we do some sort of new POP UP sub menu on click i want to point out that its STILL going to be 2 clicks, 1 to click on the menu to open up the sub menu, and then another to select the sub menu item

Thoughts?

darkdrgn2k commented:
> The layout of the 'secondary' menu seems out of place—could be a styling issue:

Maybe I should shoot this one to @Shrinks99 for suggestions since he seems to be our resident design expert lately :)

darkdrgn2k commented:
"ONCLICK" drop down menu is problematic because we will loose the ONCLICK GOTO PAGE functionality.

IE About will now drop down the submenu instead of going to about page

See About Drop Down In Sample Below
http://209.90.253.69:4000

Commit:
New Statically Displayed sub menu on mobile
Removed old mobile type menu
Made menu max height 100%

garrying committed:
tweaks

garrying committed:
Merge branch 'master' into GlobalNavigation

garrying committed:
formatting edits
